app/uk/gov/hmrc/agentpermissions/service/CustomGroupsService.scala

```diff
@@ -90,58 +92,60 @@ trait AccessGroupsService {
   taxGroupsService: TaxGroupsService,
   userClientDetailsConnector: UserClientDetailsConnector,
   auditService: AuditService
-) extends AccessGroupsService with Logging {
+) extends CustomGroupsService with Logging {
 
-  override def getById(id: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[CustomGroup]] =
-    accessGroupsRepository
+  override def getById(id: GroupId)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[CustomGroup]] =
+    customGroupsRepository
       .findById(id)
       .flatMap(withClientName)
 
   /** Gets a page of clients for the Arn and then finds if any of them are already in the group */
   override def getGroupByIdWithPageOfClientsToAdd(
-    id: String,
+    id: GroupId,
     page: Int = 1,
     pageSize: Int = 20,
     search: Option[String] = None,
     filter: Option[String] = None
   )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[(GroupSummary, PaginatedList[DisplayClient])]] =
-    accessGroupsRepository
+    customGroupsRepository
       .findById(id)
       .flatMap {
         case None => Future successful None
         case Some(grp) =>
-          val enrolmentKeys = grp.clients.map(_.map(_.enrolmentKey))
+          val enrolmentKeys = grp.clients.map(_.enrolmentKey)
           userClientDetailsConnector
             .getPaginatedClients(grp.arn)(page, pageSize, search, filter)
             .map { paginatedClients =>
               val paginatedList = PaginatedList[DisplayClient](
                 paginationMetaData = paginatedClients.paginationMetaData,
                 // TODO: NEED TO ONLY LOAD THE CLIENTS FROM THE GROUP THAT ARE IN THIS PAGINATED LIST
                 pageContent = paginatedClients.pageContent.map { c =>
-                  DisplayClient.fromClient(c, enrolmentKeys.getOrElse(Set.empty[String]).contains(c.enrolmentKey))
+                  DisplayClient.fromClient(c, enrolmentKeys.contains(c.enrolmentKey))
                 }
               )
-              Some((GroupSummary(grp._id.toString, grp.groupName, None, grp.teamMembers.size, None), paginatedList))
+              Some((GroupSummary(grp.id, grp.groupName, None, grp.teamMembers.size, None), paginatedList))
             }
       }
 
   override def create(
     accessGroup: CustomGroup
   )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[AccessGroupCreationStatus] =
-    accessGroupsRepository.get(accessGroup.arn, accessGroup.groupName) flatMap {
+    customGroupsRepository.get(accessGroup.arn, accessGroup.groupName) flatMap {
       case Some(_) =>
         Future.successful(AccessGroupExistsForCreation)
       case _ =>
         for {
           maybeCalculatedAssignments <- userEnrolmentAssignmentService.calculateForGroupCreation(accessGroup)
-          maybeCreationId            <- accessGroupsRepository.insert(withClientNamesRemoved(accessGroup))
+          maybeCreationId            <- customGroupsRepository.insert(withClientNamesRemoved(accessGroup))
```

Comment:
Not something to address right now, but will we still be removing the client names going forward if we start using `SensitiveClient`?